control-plane/tests/test_service_persistence_action_reconciliation.py

```
@@ -1444,13 +1444,16 @@ def test_service_fail_closes_when_shuffle_finalization_fails_after_dispatch(
         )
         original_persist_record = service.persist_record
 
-        def persist_record_with_finalization_failure(record: object) -> object:
+        def persist_record_with_finalization_failure(
+            record: object,
+            **kwargs: object,
+        ) -> object:
             if (
                 isinstance(record, ActionExecutionRecord)
```

Ruff (0.15.10)

[error] Undefined name `ActionExecutionRecord`

(F821)